compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -330,6 +329,34 @@ getClsDocs ctx@DocCtx{..} (DeclData (L _ (TyClD _ c@ClassDecl{..})) docs) = do
       | L loc name <- names
       ]
     f _ _ = []
+
+    -- | Remove the implied context from typeclass member functions.
+    dropMemberContext :: Maybe Anchor -> [T.Text] -> FunctionDoc -> FunctionDoc
+    dropMemberContext cl_anchor cl_args fn = fn
+        { fct_context = do
+            context <- fct_context fn
+            case context of
+                TypeTuple xs -> do
+                    let xs' = filter (not . matchesMemberContext cl_anchor cl_args) xs
+                    guard (notNull xs')
+                    Just (TypeTuple xs')
+
+                _ -> Just context
+                    -- TODO: Move to using a more appropriate type
```

Comment:
Do we expect to ever reach this case atm? If not, maybe just call `error` so we see if we mess it up instead of silently doing something that might be wrong?